flink/v1.13/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java

```diff
@@ -377,6 +378,23 @@ void createIcebergTable(ObjectPath tablePath, CatalogBaseTable table, boolean ig
     PartitionSpec spec = toPartitionSpec(((CatalogTable) table).getPartitionKeys(), icebergSchema);
 
     ImmutableMap.Builder<String, String> properties = ImmutableMap.builder();
+
+    List<WatermarkSpec> watermarkSpecs = table.getSchema().getWatermarkSpecs();
+    if (watermarkSpecs != null) {
+      for (int i = 0; i < watermarkSpecs.size(); i++) {
```

Comment:
I am not familiar with that part of Flink. can you help me understand the list of watermark spec? is it to define multiple watermark strategies and writer can pick one strategy to use?

Reply:
In flink just the first watermark strategy will be apply.

Reply:
then we need to add the validation that there is only one watermark spec defined

Reply:
@hameizi  if only one water mark is allowed why bother the for loop